packages/interfaces/src/protocols.ts

```diff
@@ -125,114 +125,156 @@ export type Callback<T extends IDecodedMessage> = (
+  INVALID_CONTENT_TOPIC = "Invalid content topic",
+  PUSH_MESSAGE_FAILED = "Push message failed",
+  EMPTY_MESSAGE = "Empty message received",
+  MISSING_PUBSUB_TOPIC = "Pubsub topic missing from push message"
+}
 
-  /**
-   * The message payload is empty, making the message invalid. Ensure that a non-empty
-   * payload is set on the outgoing message.
-   */
-  EMPTY_PAYLOAD = "Payload is empty",
+// Protocol-specific failure interfaces
+export interface LightPushFailure {
+  error: LightPushError;
+  peerId?: PeerId;
+  statusCode?: number;
+  statusDesc?: string;
+  /** Protocol version used when the failure occurred (v2, v3, etc.) */
+  protocolVersion?: string;
+}
 
-  /**
-   * The message size is above the maximum message size allowed on the Waku Network.
-   * Compressing the message or using an alternative strategy for large messages is recommended.
-   */
-  SIZE_TOO_BIG = "Size is too big",
+export interface FilterFailure {
+  error: FilterError;
+  peerId?: PeerId;
+  statusCode?: number;
+  statusDesc?: string;
+  requestId?: string;
+}
 
-  /**
-   * The PubsubTopic passed to the send function is not configured on the Waku node.
-   * Please ensure that the PubsubTopic is used when initializing the Waku node.
-   */
-  TOPIC_NOT_CONFIGURED = "Topic not configured",
+// Protocol-specific result types
+export type LightPushCoreResult = ThisOrThat<
+  "success",
+  PeerId,
+  "failure",
+  LightPushFailure
+>;
 
-  /**
-   * Fails when
-   */
-  STREAM_ABORTED = "Stream aborted",
+export type FilterCoreResult = ThisOrThat<
+  "success",
+  PeerId,
+  "failure",
+  FilterFailure
+>;
 
+export type LightPushSDKResult = ThisAndThat<
+  "successes",
+  PeerId[],
+  "failures",
+  LightPushFailure[]
+> & {
   /**
-   * General proof generation error message.
-   * nwaku: https://github.com/waku-org/nwaku/blob/c3cb06ac6c03f0f382d3941ea53b330f6a8dd127/waku/waku_rln_relay/group_manager/group_manager_base.nim#L201C19-L201C42
+   * Protocol versions used per peer during this send operation
+   * Key: peer ID string, Value: protocol version (e.g., "v2", "v3")
    */
-  RLN_PROOF_GENERATION = "Proof generation failed",
+  protocolVersions?: Record<string, string>;
+};
 
-  //
-  // RECEIVE ERRORS SECTION
-  //
-  /**
-   * The pubsub topic configured on the decoder does not match the pubsub topic setup on the protocol.
-   * Ensure that the pubsub topic used for decoder creation is the same as the one used for protocol.
-   */
-  TOPIC_DECODER_MISMATCH = "Topic decoder mismatch",
+export type FilterSDKResult = ThisAndThat<
+  "successes",
+  PeerId[],
+  "failures",
+  FilterFailure[]
+>;
 
-  /**
-   * The topics passed in the decoders do not match each other, or don't exist at all.
-   * Ensure that all the pubsub topics used in the decoders are valid and match each other.
-   */
+// Legacy types for backward compatibility (to be deprecated)
+/**
+ * @deprecated Use LightPushError or FilterError instead
+ */
+export enum ProtocolError {
```

Comment:
is it used anywhere? 
if no I would just remove it

Reply:
same for other `Failure` and `SDKProtocolResult`, etc

Reply:
ProtocolError specifically is used in quite a few places

Reply:
yes, I see

it is used in Metadata and other protocols

ok to keep it